src/mesh/region_set.cc

```diff
@@ -0,0 +1,291 @@
+/*
+ * region_set.cc
+ *
+ *  Created on: Nov 27, 2012
+ *      Author: jb
+ */
+
+#include "mesh/region_set.hh"
+
+
+namespace IT = Input::Type;
+
+
+/*******************************************************************
+ * implementation of RegionSetBase
+ */
+
+IT::Abstract & RegionSetBase::get_input_type() {
```

Comment:
This and  following three methods replace by getter for mesh->region_db_, and call its methods directly in child classes.